x-pack/plugins/apm/public/hooks/useAnomalyDetectionSettings.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useFetcher } from './useFetcher';
+
+export function useAnomalyDetectionSettings() {
+  return useFetcher(
+    (callApmApi) =>
+      callApmApi({
+        pathname: `/api/apm/settings/anomaly-detection`,
```

Comment:
In a previous PR review I asked Oliver to change this to 
```suggestion
        pathname: `/api/apm/settings/anomaly-detection/jobs`,
```

to make it obvious that it returned the jobs and to align it with `POST /api/apm/settings/anomaly-detection/jobs`.
However, he ran in to some typescript issues (it was not possible to have the same route twice even for different http methods). 
Would you mind looking into it?

Reply:
I was thinking about fixing this when we can use template string types. We could index route definitions as `POST /api/apm/settings/anomaly-detection/jobs` and `GET /api/apm/settings/anomaly-detection/jobs`.  I expect this to be easier to implement than resolving this specific issue with the options we have today. Thoughts?

Reply:
Sure, that's fine with me 👍